### PR TITLE
[cmake] add flag for SwiftCore_ENABLE_SHORT_MANGLING_LOOKUPS

### PR DESCRIPTION
**Pull request by justice-adams-apple**
add `SwiftCore_ENABLE_SHORT_MANGLING_LOOKUPS` and the associated compile definition

### CODE REVIEW

**Comment by justice-adams-apple** [comment]
@swift-ci please smoke test

File: Runtimes/Core/CMakeLists.txt [hunk 1]
```diff
@@ -91,6 +91,7 @@ defaulted_option(SwiftCore_ENABLE_COMPACT_ABSOLUTE_FUNCTION_POINTERS "Resolve ab
 defaulted_option(SwiftCore_ENABLE_BACKDEPLOYMENT_SUPPORT "Add symbols for runtime backdeployment")
 defaulted_option(SwiftCore_ENABLE_STDLIB_TRACING "Enable tracing in the runtime. Assumes the presence of os_log(3) and the os_signpost(3) API.")
 option(SwiftCore_ENABLE_UNICODE_DATA "Include unicode data in Swift runtimes" ON)
+option(SwiftCore_ENABLE_SHORT_MANGLING_LOOKUPS "Build with fast-path context descriptor lookups based on well-known short manglings." ON)
```
Comment (etcwilde) [inline Runtimes/Core/CMakeLists.txt:94]: Is this on for all platforms currently, or is this just for Apple?

**Review by etcwilde — APPROVED** [review]
Assuming that it's for all platforms, LGTM.

**Review by compnerd — APPROVED** [review]

**Comment by justice-adams-apple** [comment]
@swift-ci please smoke test

**Pull request merged**